wsmaster/che-core-api-workspace-shared/src/main/java/org/eclipse/che/api/workspace/shared/Constants.java

```diff
@@ -90,6 +90,21 @@
    */
   public static final String WORKSPACE_TOOLING_EDITOR_ATTRIBUTE = "editor";
 
+  /**
+   * The attribute allows to configure workspace to be ephemeral with no PVC attached on K8S /
+   * OpenShift infrastructure. Should be set/read from {@link WorkspaceConfig#getAttributes}.
+   *
+   * <p>Value is expected to be boolean, and if set to 'false' regardless of the PVC strategy,
+   * workspace volumes would be created as `emptyDir`. When a workspace Pod is removed for any
+   * reason, the data in the `emptyDir` volume is deleted forever
+   *
+   * @see <a
+   *     href="https://www.eclipse.org/che/docs/kubernetes-admin-guide.html#che-workspaces-storage">Che
+   *     PVC strategies</a>
+   * @see <a href="https://kubernetes.io/docs/concepts/storage/volumes/#emptydir">emptyDir</a>
+   */
+  public static final String MOUNT_SOURCES_ATTRIBUTE = "mountSources";
```

Comment:
1. I was not able to find anything similar for docker infra. `VolumesConverter` just grabs all the volumes from workspace config and adds them to container config.

2. yeah, this is a good point. Maybe `persistVolumes` would be a better option ? @l0rd @garagatyi wdyt ?

Reply:
1. I think we don't have to support this in Docker infra. It can be implemented later in case it would be needed.
2. `persistVolumes` looks good to me

Reply:
@sleshchenko attribute has been renamed to `persistVolumes`